CI no longer works due to missing conda via micromamba

See eg https://github.com/karel-brinda/MiniPhy/actions/runs/14202700238/job/39793155493?pr=110

```
Run micromamba activate test-env
if [ -d ".test" ]; then \
	make -C .test test; \
else\
	snakemake --cores all --use-conda --conda-prefix=".conda" --rerun-incomplete --printshellcmds ; \
fi
make[1]: Entering directory '/Users/runner/work/MiniPhy/MiniPhy/.test'
if [ -d ".test" ]; then \
	make -C .test test; \
else\
	snakemake --cores all --use-conda --conda-prefix=".conda" --rerun-incomplete --printshellcmds ; \
fi
Assuming unrestricted shared filesystem usage for local execution.
Building DAG of jobs...
/bin/bash: conda: command not found
Traceback (most recent call last):
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/cli.py", line 2014, in args_to_api
    dag_api.execute_workflow(
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/api.py", line 5[9](https://github.com/karel-brinda/MiniPhy/actions/runs/14202700238/job/39793155493?pr=110#step:5:10)0, in execute_workflow
    workflow.execute(
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/workflow.py", line 1114, in execute
    self.dag.create_conda_envs()
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/dag.py", line 418, in create_conda_envs
    env.create(self.workflow.dryrun)
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/deployment/conda.py", line 393, in create
    pin_file = self.pin_file
               ^^^^^^^^^^^^^
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake_interface_common/utils.py", line 33, in __get__
    value = self.method(instance)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/deployment/conda.py", line [10](https://github.com/karel-brinda/MiniPhy/actions/runs/14202700238/job/39793155493?pr=110#step:5:11)3, in pin_file
    f".{self.conda.platform}.pin.txt"
        ^^^^^^^^^^
  File "/Users/runner/micromamba/envs/test-env/lib/python3.[11](https://github.com/karel-brinda/MiniPhy/actions/runs/14202700238/job/39793155493?pr=110#step:5:12)/site-packages/snakemake_interface_common/utils.py", line 33, in __get__
    value = self.method(instance)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/deployment/conda.py", line 96, in conda
    return Conda(
           ^^^^^^
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/deployment/conda.py", line 653, in __init__
    shell.check_output(self._get_cmd("conda info --json"), text=True)
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/site-packages/snakemake/shell.py", line 61, in check_output
    return sp.check_output(cmd, shell=True, executable=executable, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/micromamba/envs/test-env/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'conda info --json' returned non-zero exit status [12](https://github.com/karel-brinda/MiniPhy/actions/runs/14202700238/job/39793155493?pr=110#step:5:13)7.
make[1]: *** [Makefile:101: test] Error 1
make[1]: Leaving directory '/Users/runner/work/MiniPhy/MiniPhy/.test'
make: *** [Makefile:101: test] Error 2
Error: Process completed with exit code 2.
```

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -107,7 +107,7 @@ jobs:
           conda --version
           conda info --json
           rm -rf .test/.conda .conda
-          make test
+          make test CONDA_FRONTEND=conda
 #      - name: Make bigtest
 #        shell: bash -el {0}
 #        run: |

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -76,9 +76,8 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-      - uses: mamba-org/setup-micromamba@v1
+      - uses: mamba-org/setup-micromamba@v3
         with:
-          micromamba-version: "1.5.6-0"
           environment-name: test-env
           condarc: |
             channels:

--- a/Makefile
+++ b/Makefile
@@ -19,8 +19,10 @@ ifeq ($(USE_CONDA),)
     $(error 'use_conda' not found in the configuration)
 endif
 
+CONDA_FRONTEND ?= $(shell command -v mamba >/dev/null 2>&1 && echo mamba || echo conda)
+
 ifeq ($(strip $(USE_CONDA)),True)
-	CONDA_PARAMS  =	--use-conda --conda-prefix="$(CONDA_DIR)"
+	CONDA_PARAMS = --use-conda --conda-frontend $(CONDA_FRONTEND) --conda-prefix="$(CONDA_DIR)"
 endif
 
 ifeq ($(SMK_CLUSTER_ARGS),)

